Places equipped grear in excludes automatically when toggle is set.

JonHinkerton opened the pull request:
If gui is closed and dollroom is open, adding a piece of equipment will add it automatically to the exclude list while the toggle is set, preventing unwanted changes to custom outfits. If gui is open regardless of toggle, then include/exclude behavior should be as normal.

JonHinkerton committed:
Places equipped grear in excludes automatically when toggle is set.

xADDBx commented on ReDress/Main.cs:
```diff
@@ -301,6 +308,11 @@ static void OnGUI(UnityModManager.ModEntry modEntry) {
+
+                    DrawDiv();
+                    excludeNewEESave = GUILayout.Toggle(excludeNewEESave, "Automatically exclude new items on all characters");
+                    excludeNewEE = false; //disable while gui is open
```
OnShowGUI is a thing in UnityModManager

xADDBx commented on ReDress/Main.cs:
```diff
@@ -104,6 +108,8 @@ static void OnHideGUI(UnityModManager.ModEntry modEntry) {
         colorPickerItem = "";
         selectedOutfit = Outfit.Current;
+        excludeNewEE = excludeNewEESave;
+        settings.Save(modEntry);
```
Are you sure you need to manually call that here?

xADDBx commented:
I'll take a proper look at a later time since I'll likely be a little busy for the next few days. If I don't respond until Friday feel free to ping me on Discord; I might've forgotten then.

xADDBx committed:
w

.

-

UnityModManager saves properly as long as the game doesn't crash. Adding an extra call here is unnecessary

Smallish adjustments

xADDBx committed:
Smallish adjustments

xADDBx committed:
UnityModManager saves properly as long as the game doesn't crash. Adding an extra call here is unnecessary

xADDBx committed:
-

xADDBx committed:
.

xADDBx committed:
w

xADDBx commented:
@JonHinkerton if you can confirm this still works as intended I'll merge

JonHinkerton committed:
Merge branch 'autoExemptNew' of https://github.com/JonHinkerton/ReDress-WH40K-Rogue-Trader into autoExemptNew

JonHinkerton committed:
Fix inventory window open conditional

JonHinkerton commented:
UMM state doesn't need to be tracked, only the inventory window because it's trying to exclude manual gear swaps only.

xADDBx commented:
So the current version works?

JonHinkerton commented:
The squash for your tweaks plus the new commit to remove the umm tracking and conditional works for me.
